### ISSUE TITLE
BWEM sometimes has no areas

### ISSUE THREAD

**Issue by jordanfly** [JavaBWAPI/JBWAPI#34]
I've been having an intermittent crashing problem. It's very rare, occurring in about 1/10 or 1/20 games. I've tried to put together some information that might be helpful.

If the crash is going to occur in a game, it will always occur the first time a unit attempts to pathfind. There will be a 1-2 second lag spike, followed by the following information in the console window:

java.lang.IllegalStateException
	at bwem.BWMap.breadthFirstSearch(BWMap.java:333)
	at bwem.BWMap.breadthFirstSearch(BWMap.java:339)
	at bwem.Graph.getNearestArea(Graph.java:79)
	at bwem.Graph$Pathing.<init>(Graph.java:697)
	at bwem.Graph.getPath(Graph.java:154)
	at bwem.BWMap.getPath(BWMap.java:220)
	at SimplicityBot$PositionPath.autogeneratePath(SimplicityBot.java:4414)
	at SimplicityBot$UnitCtrl.orders_pathfind(SimplicityBot.java:4794)
	at SimplicityBot$UnitCtrl.orders_workerPremove(SimplicityBot.java:5403)
(and so on, the call stack goes back through a bunch more of my stuff.)

I have confirmed that I'm passing two valid positions to BWMap.getPath.

What caught my attention is the call from getNearestArea to breadthFirstSearch. GetNearestArea first checks if the position is in an area, and if so, it just returns that area. See Graph.java at line 92. It only calls breadthFirstSearch if the position is not located in an area. But it should be: the crash here occurred when I tried to premove a worker to my natural to build a hatchery; the start position is the worker and the end position is the base location he's headed to. Those should both be within normal areas. 

I added the following testing code to my onFrame method: 
```
int areas = bwem.getMap().getAreas().size();
if (areas == 0) printLine("BWEM areas array size 0");
```

The next time the crash occurred, it was only after spamming the console with hundreds of "BWEM areas array size 0." So I think that's the cause of the crash. This most recent occasion I was playing on the top-right corner of Empire of the Sun from the SSCAIT map pack.

I would like to know if I'm setting something up wrong or if there is a problem somewhere. I would greatly appreciate any help!

I'm not sure if there is a consistent setup to cause the crash. I rarely see it, maybe 1/10 games or so. One possibility is that I don't always restart Starcraft after every game. I usually leave the game up and only restart my bot. I wonder if some data from a prior game could be lingering around or causing a problem. I can't say for sure whether the crash has ever happened on a fresh instance of Starcraft. I will try testing that next.

Thank you very much!

**Comment by jordanfly** [comment]
I apologize; after further testing I think it happens because of my setup. 

Sometimes I click the button to restart my bot when the scores display is still up. I think that's the problem. When I did that again just now and changed some other settings in setting up the game, I was able to produce other crashes. So I think something in my setup is the problem. Sorry for any confusion I'm causing here!
